objects/Door.js

```diff
@@ -0,0 +1,74 @@
+const DOOR_TRIGGERS = {
+    AUTOMATIC: 'auto',
+    UP: 'up',
+    DOWN: 'down'
+}
+class Door extends FinishFlag {
+    constructor(x, y, tileSize, type, tilemapHandler, extraAttributes = {}) {
+        super(x, y, tileSize, type, tilemapHandler, extraAttributes);
+        
+        // overrides
+        this.lockedSpriteIndex = SpritePixelArrays.getIndexOfSprite(ObjectTypes.LOCKED_DOOR);
+        this.lockedSpriteYPos = this.lockedSpriteIndex * this.tileSize;
+        this.sound = SoundHandler.door;
+
+        // new
+        if(!extraAttributes?.activationTrigger) {
+            this.addChangeableAttribute("activationTrigger", DOOR_TRIGGERS.UP);
+        }
+        if(!extraAttributes?.flagIndex) {
+            this.flagIndex = this.makeid(3);
+            this.addChangeableAttribute("flagIndex", this.flagIndex, this.tilemapHandler.currentLevel);
+        }
+    }
+
+    collisionEvent() { // overrides super.collisionEvent
+        if (this.locked) {
+            return;
+        }
+        PlayerInteractionHandler.registerInteractionTarget(this);
+    }
+
+    changeExit(customExit, reciprocatingLevelIndex) {
+        super.changeExit(customExit, reciprocatingLevelIndex);
+        
+        // Door-specific effect: if exit target is a door, connect it back to this
+        // (make the connection mutual)
+        let targetLevelIndex = customExit.levelIndex;
+        let targetObjectType = customExit.type;
+        let targetFlagIndex = customExit.flagIndex;
+
+        if (targetObjectType !== ObjectTypes.DOOR) {
+            return;
+        }
+        if (!reciprocatingLevelIndex) {
+            // (in this case, this door is probably the one who is reciprocating)
+            return;
+        }
```

Comment:
```suggestion
        if (targetObjectType !== ObjectTypes.DOOR || !reciprocatingLevelIndex) {
            return;
        }
```

Reply:
Ok. This is another style/readability choice. Maybe style guide entry number 3 would be something like "Consolidate conditionals." I feel like there should be a length limit before finding a good place to line-break or something though. Long, line-wrapping conditionals are like run-on sentences and can take a bit of dedication to correctly untangle.